### PR TITLE
Support Rust `no_std` environments

### PR DESCRIPTION
**Pull request by DaniPopes**
Add a "std" feature flag to the Rust bindings, enabled by default, which when disabled by consumers with `no-default-features = true` activates `#![no_std]` attribute.

This is achieved by using `alloc` for owned `String` and `Vec`, and using `&CStr` instead of `&Path` when `std` is not available.

I've also included some small warnings and lint fixes.

### CODE REVIEW

**Review by jtraglia — APPROVED** [review]
Nice, I like these changes. Thanks! Just one thing I'd like to point out (see my comment below), but it's not your problem. I'd like to have @asn-d6 and @pawanjay176 also review this before merging.

File: bindings/rust/src/bindings/test_formats/verify_blob_kzg_proof_batch.rs [hunk 1]
```diff
@@ -12,11 +14,7 @@ pub struct Input {
 
 impl Input {
     pub fn get_blobs(&self) -> Result<Vec<Blob>, Error> {
-        let mut v: Vec<Blob> = Vec::new();
-        for blob in &self.blobs {
-            v.push(Blob::from_hex(blob)?);
-        }
-        return Ok(v);
+        self.blobs.iter().map(|blob| Blob::from_hex(blob)).collect()
```
[the remark this answers is not in the document]
Comment (DaniPopes) [reply bindings/rust/src/bindings/test_formats/verify_blob_kzg_proof_batch.rs:17]: Looks like it's failing in CI as well, will revert

**Review by pawanjay176 — COMMENTED** [review]
LGTM overall. 
The feature checks need to set the working directory and also the `no-std` check doesn't compile on the tests

**Comment by DaniPopes** [comment]
Should be good now

**Comment by jtraglia** [comment]
@DaniPopes Ah sorry, that windows stack overflow issue is unrelated to your PR. We just fixed this. See:
 
* #331 
* #354 

You just need to pull in changes from main. Thanks for your work on this 🥇

File: .github/workflows/rust-tests.yml [hunk 2]
```diff
@@ -8,6 +8,18 @@ on:
       - main
 
 jobs:
+  feature-checks:
+    runs-on: ubuntu-latest
+    timeout-minutes: 30
+    steps:
+      - uses: actions/checkout@v3
```
Comment (jtraglia) [inline .github/workflows/rust-tests.yml:15]: I think doing a recursive checkout will fix the feature checks issue.
```suggestion
      - uses: actions/checkout@v3
        with:
          submodules: recursive
```

**Pull request merged**